Sleep Tracker/Sleep Tracker/View/SleepButton.swift

```diff
@@ -0,0 +1,43 @@
+import SwiftUI
+
+/// A view that displays a button to control sleep.
+struct SleepButton: View {
+    // MARK: - Public interface
+
+    /// A string value that contains the text on the button. Empty string by default.
+    @State var buttonText = String()
+    /// A Boolean value that indicates whether the text on the button should change after it is tapped.
+    /// False by default.
+    @State var shouldChangeButtonText = false
+    /// A string value that contains the text on the button after it is tapped. Empty string by default.
+    @State var newButtonText = String()
+
+    var body: some View {
+        Button(action: {
+            changeButtonText(to: newButtonText)
+        }, label: {
+            RoundedRectangle(cornerRadius: 46)
+                .foregroundColor(.AppScheme.brightBlue)
+                .frame(width: 338, height: 54)
+                .overlay(content: {
+                    Text(buttonText)
+                        .foregroundColor(.AppScheme.white)
+                        .font(.system(size: 20, weight: .bold))
+                })
+        })
```

Comment:
Can you simplify the implementation and make it more flexible?

Your values of the frame means the width of the button will be the same on a different screens (iPhones). Is that what you wanted?

Something like that:
```suggestion
        Button(action: {
            changeButtonText(to: newButtonText)
        }, label: {
            Text(buttonText)
                .foregroundColor(.AppScheme.white)
                .font(.system(size: 20, weight: .bold))
                .frame(maxWidth: .infinity)
                .padding()

        })
        .background(Color.AppScheme.brightBlue)
        .cornerRadius(46)
        .padding(24)
```
?

Reply:
No, that's not what I wanted.

Reply:
Fixed.